src/future-attribute.test.js

```diff
@@ -0,0 +1,100 @@
+import assert from 'assert';
+import sinon from 'sinon';
+
+import _ from 'underscore';
+import { Model } from 'backbone';
+
+import { when, whenever } from './future-attribute.js';
+
+function callbackChecker(model, value, options, context, done) {
+    return sinon.fake(function(m, v, o) {
+        assert(m === model);
+        assert(_.contains(value, v));
+        assert.deepStrictEqual(o, options);
+        assert(this === context);
+        done();
+    });
+}
+
+describe('when', function() {
+    it('invokes the callback if the attribute is present', function(done) {
+        var model = new Model({a: 1});
+        var spy = callbackChecker(model, [1], {}, model, done);
+        when(model, 'a', spy);
+        assert(spy.notCalled);
```

Comment:
No, the test is correct. The spy is eventually called, but async. Hence, it should *not yet* be called at the time of line 24.

The spec itself is also async (because of the `done` argument) and it will only finish and pass after the spy has been invoked. This is detected because the spy is created by the `callbackChecker` function, which ensures that the spy finished by calling `done`.